sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/msfaa-integration/_tests_/msfaa-full-time-process-response-integration.scheduler.e2e-spec.ts

```diff
@@ -0,0 +1,231 @@
+      const { nestApplication, dataSource, sshClientMock } =
+        await createTestingAppModule();
+      app = nestApplication;
+      db = createE2EDataSources(dataSource);
+      sftpClientMock = sshClientMock;
+      // Processor under test.
+      processor = app.get(FullTimeMSFAAProcessResponseIntegrationScheduler);
+    });
+
+    beforeEach(async () => {
+      jest.clearAllMocks();
+      // Force any not signed MSFAA to be signed to ensure that new
+      // ones created will be the only ones available to be updated.
+      await db.msfaaNumber.update(
+        { dateSigned: IsNull() },
+        { dateSigned: getISODateOnlyString(new Date()) },
+      );
+      // Cancel any not canceled MSFAA.
+      await db.msfaaNumber.update(
+        { cancelledDate: IsNull() },
+        { cancelledDate: getISODateOnlyString(new Date()) },
+      );
+    });
+
+    it("Should reactivate a cancelled MSFAA when the same MSFAA is received in the response file and re-associate this reactivated MSFAA with all pending disbursements.", async () => {
+      // Arrange
+      // Create a cancelled MSFAA. The Msfaa number used for creating the cancelled Msfaa record is the same as the one used in the msfaa-full-time-file-with-reactivation-record.dat
+      // Ensuring that any previous runs of this test or any other test do not have the same Msfaa id as the one in the re-activation file.
+      await db.msfaaNumber.update(
+        { msfaaNumber: FULL_TIME_SAMPLE_MSFAA_NUMBER },
+        { msfaaNumber: THROW_AWAY_MSFAA_NUMBER },
+      );
+      const student = await saveFakeStudent(db.dataSource);
+      const cancelledMSFAARecord = await db.msfaaNumber.save(
+        createFakeMSFAANumber(
+          { student },
+          {
+            msfaaState: MSFAAStates.CancelledOtherProvince,
+          },
+          {
+            msfaaNumber: FULL_TIME_SAMPLE_MSFAA_NUMBER,
+            offeringIntensity: OfferingIntensity.fullTime,
+          } as MSFAANumber,
+        ),
+      );
+      // Create Pending MSFAA and associate it with disbursements from the two applications.
+      const currentMSFAA = createFakeMSFAANumber(
+        { student },
+        {
+          msfaaState: MSFAAStates.Pending,
+        },
+        {
+          offeringIntensity: OfferingIntensity.fullTime,
+        },
+      );
+      await db.msfaaNumber.save(currentMSFAA);
+      // Create 2 applications with 4 disbursements - 2 pending state and 2 sent state.
+      const applicationA = await saveFakeApplicationDisbursements(
+        db.dataSource,
+        {
+          student,
+          msfaaNumber: currentMSFAA,
+        },
+        {
+          applicationStatus: ApplicationStatus.Completed,
+          offeringIntensity: OfferingIntensity.fullTime,
+          createSecondDisbursement: true,
+          disbursementScheduleStatusA: DisbursementScheduleStatus.Sent,
+        },
+      );
+      const applicationB = await saveFakeApplicationDisbursements(
```

Comment:
Do we really need 2 applications?

One application with 2 disbursements(firstDisbursement-Sent, seconfDisBursement-Pending) will be sufficient right